src/API.jl

```diff
@@ -1469,11 +1469,11 @@ function precompile(ctx::Context, pkgs::Vector{PackageSpec}; internal_call::Bool
                         end
                         loaded && (n_loaded += 1)
                     catch err
-                        close(iob)
+                        close(std_pipe.in) # close pipe to end the std output monitor
                         wait(t_monitor)
```

Comment:
Why do these only happen on the error path?

Reply:
Good point.